cmd/query/app/http_handler_test.go

```diff
@@ -388,6 +404,30 @@ func TestGetTraceBadTraceID(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestGetTraceBadTimeWindow(t *testing.T) {
+	testCases := []struct {
+		name  string
+		query string
+	}{
+		{
+			name:  "Bad start time",
+			query: "start=a",
+		},
+		{
+			name:  "Bad end time",
+			query: "end=b",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ts := initializeTestServer(t)
+			var response structuredResponse
+			err := getJSON(ts.server.URL+`/api/traces/123456?`+tc.query, &response)
+			require.Error(t, err)
```

Comment:
this is non-specific test, it can succeed for the other reasons than the condition you're testing. It's better to add `err string` to the test cases and check `require.ErrorContains`

Reply:
done